custom_libs/subliminal_patch/subtitle.py

```diff
@@ -65,6 +66,11 @@ class Subtitle(Subtitle_):
     # format = "srt" # default format is srt
 
     def __init__(self, language, hearing_impaired=False, page_link=None, encoding=None, mods=None, original_format=False):
+        # language needs to be cloned because it is actually a reference to the provider language object
+        # if a new copy is not created then all subsequent subtitles for this provider will incorrectly be modified
+        # at least until Bazarr is restarted or the provider language object is recreated somehow
+        language = deepcopy(language)
+
         # set subtitle language to hi if it's hearing_impaired
         if hearing_impaired:
             language = Language.rebuild(language, hi=True)
```

Comment:
This is really just a though, would this work?

```suggestion
    language = Language.rebuild(language, hi=hearing_impaired)
```

Reply:
What you think @vitiko98?

Reply:
I think it *might* work, but what’s the benefit?
You would always be forcing a Language.rebuild(), even when it’s not indicated.
You would be doing the deepcopy() presumably as a side-effect of the rebuild() but I feel that would obfuscate the fact that we don’t want to modify the provider object. That may lead to someone reorganizing the code in the future and unintentionally recreating the same bug.

Reply:
It was really a though as it seems we always want a new instance of the language, and im not sure, the deep copy looks like a workaround.

Reply:
Honestly, I don’t think it’s a workaround, but the way it should’ve been from the beginning. We can’t be passing global objects around and having them be modified unintentionally. It should’ve always been a copy right from the beginning.
I’m also wondering if this is the potential reason for other mysterious bugs that we’ve never been able to solve.
Let’s commit this and see what happens.